decidim-admin/app/models/decidim/admin/abilities/participatory_process_admin_user.rb

```diff
@@ -9,55 +9,45 @@ module Abilities
       # Admin engine.
       #
       # This ability will not apply to organization admins.
-      class ParticipatoryProcessAdmin
-        include CanCan::Ability
-
-        def initialize(user, _context)
-          @user = user
-
-          return unless user && !user.role?(:admin) && !user.role?(:collaborator)
-
-          can :read, :admin_dashboard do
-            participatory_processes.any?
-          end
+      class ParticipatoryProcessAdminUser < Decidim::Abilities::ParticipatoryProcessAdminUser
+        def define_abilities
+          super
 
           can :manage, ParticipatoryProcess do |process|
-            participatory_processes.include?(process)
+            can_manage_process?(process)
           end
 
           cannot :create, ParticipatoryProcess
           cannot :destroy, ParticipatoryProcess
+        end
 
-          cannot :manage, :admin_users
+        def define_participatory_process_abilities
+          super
 
-          can :manage, ParticipatoryProcessUserRole do |role|
-            role.user != user
+          can :manage, Feature do |feature|
+            can_manage_process?(feature.participatory_process)
           end
 
-          can :manage, Moderation do |moderation|
-            participatory_processes.include?(moderation.participatory_process)
+          can :manage, Category do |category|
+            can_manage_process?(category.participatory_process)
           end
 
           can :manage, Attachment do |attachment|
-            participatory_processes.include?(attachment.attached_to)
+            can_manage_process?(attachment.attached_to)
```

Comment:
Yeah, it is polymorphic. I'll add `attached_to.kind_of? ParticipatoryProcess` then.